packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -161,57 +165,81 @@ export const generateTypeDefGlobalContext = () => {
-  type GenerateResponse = { filename: string; contents: string }[]
+export const generateTypeDefGraphQLApi = async () => {
   try {
-    // TODO: Figure out how to get this type.
-    const generates: Record<string, unknown> = {}
-
-    if (['api', 'all'].includes(side)) {
-      generates[path.join(rwjsPaths.api.base, 'types/graphql.d.ts')] = {
-        plugins: ['typescript', 'typescript-resolvers'],
-      }
-    }
-    if (['web', 'all'].includes(side) && findCells().length) {
-      generates[path.join(rwjsPaths.web.base, 'types/graphql.d.ts')] = {
-        documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
-        plugins: ['typescript', 'typescript-operations'],
-      }
-    }
-
-    // https://www.graphql-code-generator.com/docs/getting-started/programmatic-usage#using-the-cli-instead-of-core
-    const f: GenerateResponse = await generate(
-      {
-        cwd: rwjsPaths.base,
-        schema: rwjsPaths.generated.schema,
-        config: {
-          scalars: {
-            DateTime: 'string',
-            Date: 'string',
-            JSON: 'Record<string, unknown>',
-            JSONObject: 'Record<string, unknown>',
-            Time: 'string',
+    const rwjsPaths = getPaths()
+    const f = await generateTypeDefGraphQL({
+      [path.join(rwjsPaths.api.base, 'types/graphql.d.ts')]: {
+        plugins: [
+          {
+            typescript: { enumsAsTypes: true },
           },
-          omitOperationSuffix: true, // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc.
-        },
-        // @ts-expect-error - Meh.
-        generates,
-        silent: false,
-        errorsOnly: true,
+          'typescript-resolvers',
+        ],
       },
-      true
-    )
-    return f.map(({ filename }) => filename)
+    })
+    return f
   } catch (e) {
-    // `generate` outputs errors which are helpful.
-    // This tries to clean up the output of those errors.
     console.error()
-    console.error('Error: Could not generate GraphQL type definitions')
+    console.error('Error: Could not generate GraphQL type definitions (api)')
     console.error()
+    return []
+  }
+}
 
+export const generateTypeDefGraphQLWeb = async () => {
+  if (findCells().length) {
+    const rwjsPaths = getPaths()
+    try {
+      const f = await generateTypeDefGraphQL({
+        [path.join(rwjsPaths.web.base, 'types/graphql.d.ts')]: {
+          documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
+          plugins: [
+            {
+              typescript: {
+                enumsAsTypes: true,
+              },
+            },
+            'typescript-operations',
+          ],
+        },
+      })
+      return f
+    } catch (e) {
+      console.error()
+      console.error('Error: Could not generate GraphQL type definitions (web)')
+      console.error()
+      return []
+    }
+  } else {
     return []
   }
 }
+
+const generateTypeDefGraphQL = async (generates: Record<string, unknown>) => {
```

Comment:
Do we still need this function?

Reply:
Yup, this contains the majority of the configuration. The other two functions pass their own small samples of configuration in here.